question18/main.js

```diff
@@ -0,0 +1,191 @@
+'use strict';
+const imgListsWrapper = document.getElementById('js-imgListsWrapper');
+const imgLists = document.getElementById('js-imgLists');
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const showLoadImg = () => {
+  const loadImg = createElementWithClassName("img", "loading");
+  loadImg.src = "./img/loading-circle.gif";
+  loadImg.id = 'loading';
+  imgListsWrapper.appendChild(loadImg);
+} 
+
+const removeLoadImg = () => {
+  document.getElementById('loading').remove();
+} 
+
+const getData = new Promise((resolve) => {
+  showLoadImg();
+  setTimeout(() => {
+    resolve(fetch('https://mocki.io/v1/09593e78-80ba-4765-940e-3c70fb46078c'));
+  }, 3000);
+})
+
+//Get json data
+const callApi = async() => {
+  try {
+    const response = await getData;
+    if (!response.ok) {
+      console.error(`${response.status}:${response.statusText}`);
+    }
+    const json = await response.json();
+    return json.data;
+  }
+  catch(error) {
+    console.error(error);
+  }
+  finally {
+    removeLoadImg();
+  }
+}
+
+const init = async() => {
+  const imgData = await callApi();
+  if (imgData){
+    renderImgUiElement(imgData);
+    } else {
+    imgListsWrapper.textContent = "データの表示に失敗しました";
+    console.error('データの表示に失敗しました'); 
+  }
+}
+init();
+
+
+const renderImgUiElement = (imgData) => {
+  renderNextBtn(imgData);
+  renderPrevBtn(imgData);
+  renderPageNumElement(imgData);
+  renderPagenation(imgData);
+  clickedPagenation(imgData);
+  autoImgSwitch(imgData);
+  imgLists.appendChild(getImgListsFragment(imgData));
+}
+
+const renderPageNumElement = (imgData) => {
+  const numberOfPage = createElementWithClassName("p", "pageNumber");
+  imgListsWrapper.appendChild(numberOfPage);
+  numberOfPage.id = "js-number";
+  numberOfPage.textContent = `1 / ${imgData.length}`;  
+}
+
+
+const getImgListsFragment = (imgData) => {
+  const fragmentImglists = document.createDocumentFragment();
+  for (let i = 0; i < imgData.length; i++) {
+    const imgList = createElementWithClassName("li", "imgList");
+    const img = createElementWithClassName("img", `img_0${[i + 1]}`);
+    imgList.dataset.index = i;
+    img.src = imgData[i].img;
+    img.alt = imgData[i].description;
+    i === 0 && imgList.classList.add("is-show");
+    fragmentImglists.appendChild(imgList).appendChild(img);
+  }
+  return fragmentImglists;
+};
+
+
+let imgNum = 0;
```

Comment:
IMO
How about assigning to an object using const instead of using let outside the function?
```suggestion
const ingNum = { count : 0 };
```

```js
// At the caller
//before
//imgNum -= 1;

ingNum.count -= 1;
```

Reply:
Thank you for your review! I learned a lot because I can't come up with it on my own. I thought it would be better to use this one because the use of const prevents unexpected reassignment. I fixed it so please confirm.
fixed commit : https://github.com/csakyo/homework/pull/32/commits/16b86688e662a216b2c0997bc8f24038cded15ca